src/xrpld/app/paths/detail/StepChecks.h

```diff
@@ -52,6 +52,13 @@ checkFreeze(
         {
             return terNO_LINE;
         }
+        if (view.rules().enabled(featureDeepFreeze))
+        {
+            if (sle->isFlag(lsfHighDeepFreeze) || sle->isFlag(lsfLowDeepFreeze))
+            {
+                return terNO_LINE;
+            }
+        }
```

Comment:
In the original code, the `if (sle->isFlag((dst > src) ? lsfHighFreeze : lsfLowFreeze))` is used to check whether the high or low freeze flag is set based on whether the `dst` Account ID is numerically larger or smaller than the `src` Account ID.

In the newly added code, since there are differences in behavior between which party (account) sets the deep freeze flag, shouldn't the same logic be used here as above, i.e. `if (sle->isFlag((dst > src) ? lsfHighDeepFreeze : lsfLowDeepFreeze))`?

Or, since the spec states:

> When an issuer enacts Deep-freeze, the following rules apply to the tokens in that trust line:
> *  ...
> * The counterparty can neither send nor receive from others on the deep-frozen trustline
> * ...

and:

> An individual address can deep-freeze its trust line to an issuer or financial institution ...
> * It prevents other addresses from sending that financial institution's tokens to the individual address.
> * It also prevents the individual address from sending the token to other non-issuer addresses.

then if either party deep freezes the trust line then `terNO_LINE` should be returned, in which case the current code is fine as-is?

Reply:
applying the same logic could work. But i don't believe that's necessary because, if _either_ low or high deep freeze flag is enabled, the trustline balance cannot be changed regardless which party enacted the deepfreeze(excluding direct payment between issuer and holder)

Reply:
I'll share a comment I wrote elsewhere on this topic:
> I'm actually having a hard time wrapping my head around the idea that if either deep freeze flag is set, then the line is deep frozen.
> The way I'm trying to reconcile it is that the only reason for the two flags is to be able to keep track of which side of the trust line set it, and thus can clear it.

It would probably be worth adding an explicit comment here and anywhere else relevant that, unlike regular freeze (or really any of the other flags), a deep frozen trustline acts the same regardless of which side froze it. The only reason for two flags is to track which side(s) froze the line.